lib/chef/provisioning/aws_driver/aws_resource.rb

```diff
@@ -133,13 +133,14 @@ def self.aws_sdk_type(sdk_class,
                         load_provider: true,
                         id: :name,
                         aws_id_prefix: nil)
+    # TODO less than Chef 12.4, this is necessary.  Above that, this is deprecated.
     self.resource_name = self.dsl_name
```

Comment:
Unrelated to this PR, but maybe we guard it with `if self.respond_to?(:resource_name)`?